hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -255,6 +256,31 @@ public class StreamSync implements Serializable, Closeable {
 
   private final boolean useRowWriter;
 
+  @VisibleForTesting
+  StreamSync(HoodieStreamer.Config cfg, SparkSession sparkSession,
```

Comment:
is there any way to consolidate the constructors so that there is one constructor with all of the args and the others just invoke `this(...)`?

Reply:
I would rather not change the actual constructor just to make the testing constructor cleaner

Reply:
The idea here is to make sure that the constructors are similar so you test a class that mirrors what is going to happen in production. If there is too much divergence between the constructors, we may think we've tested a path that is untested. In this case it's probably fine since we have a lot of functional tests around the StreamSync functionality.